### PR TITLE
Add max_tasks_per_node setting to Batch Pool

### PR DESCRIPTION
**Pull request by stuartleeks**

### CODE REVIEW

File: azurerm/resource_arm_batch_pool.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ func resourceArmBatchPool() *schema.Resource {
 				ForceNew:         true,
 				DiffSuppressFunc: suppress.CaseDifference,
 			},
+			"max_tasks_per_node": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
```
Comment (metacpp) [inline azurerm/resource_arm_batch_pool.go]: Can you add a validation for this field ?

File: azurerm/resource_arm_batch_pool.go [hunk 2]
```diff
@@ -243,6 +248,8 @@ func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error
 	poolName := d.Get("name").(string)
 	displayName := d.Get("display_name").(string)
 	vmSize := d.Get("vm_size").(string)
+	maxTasksPerNode := d.Get("max_tasks_per_node").(int)
```
Comment (metacpp) [inline azurerm/resource_arm_batch_pool.go]: Can we directly assert it as `int32` ?

File: azurerm/resource_arm_batch_pool.go [hunk 3]
```diff
@@ -243,6 +248,8 @@ func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error
...
+	foo := int32(maxTasksPerNode)
```
Comment (metacpp) [inline azurerm/resource_arm_batch_pool.go]: Can we use a more readable name ? For example, `maxTasks`.

File: azurerm/resource_arm_batch_pool.go [hunk 4]
```diff
@@ -259,8 +266,10 @@ func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error
 
 	parameters := batch.Pool{
 		PoolProperties: &batch.PoolProperties{
-			VMSize:      &vmSize,
-			DisplayName: &displayName,
+			VMSize:          &vmSize,
+			DisplayName:     &displayName,
+			MaxTasksPerNode: &foo,
+			// MaxTasksPerNode: &maxTasksPerNode,
```
Comment (metacpp) [inline azurerm/resource_arm_batch_pool.go]:
```suggestion
```

File: website/docs/d/batch_pool.html.markdown [hunk 5]
```diff
@@ -43,6 +43,8 @@ The following attributes are exported:
 
 * `start_task` - A `start_task` block that describes the start task settings for the Batch pool.
 
+* `max_tasks_per_node` - The maximum number of tasks that can run concurrently on a single compute node in the pool.
```
Comment (metacpp) [inline website/docs/d/batch_pool.html.markdown]: Can we update the documentation for the resource ?

File: azurerm/resource_arm_batch_pool.go (hunk 3 above)
Comment (stuartleeks) [reply azurerm/resource_arm_batch_pool.go]: Oops  - I meant to use a clearer name here! Done in latest commits

File: azurerm/resource_arm_batch_pool.go (hunk 4 above)
Comment (stuartleeks) [reply azurerm/resource_arm_batch_pool.go]: Done in latest commits

File: website/docs/d/batch_pool.html.markdown (hunk 5 above)
Comment (stuartleeks) [reply website/docs/d/batch_pool.html.markdown]: Missed adding the file when making the commit - sorry! Done now

File: azurerm/resource_arm_batch_pool.go (hunk 1 above)
Comment (stuartleeks) [reply azurerm/resource_arm_batch_pool.go]: Done

File: azurerm/resource_arm_batch_pool.go (hunk 2 above)
Comment (stuartleeks) [reply azurerm/resource_arm_batch_pool.go]: I've removed the temporary variable now!

**Comment by stuartleeks** [comment]
@metacpp - Thanks for your review. I've just pushed some changes to address your feedback. Let me know if you have any further comments.

File: website/docs/r/batch_pool.html.markdown [hunk 6]
```diff
@@ -102,6 +102,8 @@ The following arguments are supported:
 
 * `display_name` - (Optional) Specifies the display name of the Batch pool.
 
+* `max_tasks_per_node` - (Optional) Specifies the maximum number of tasks that can run concurrently on a single compute node in the pool.
```
Comment (katbyte) [inline website/docs/r/batch_pool.html.markdown]: Could we add the default value here? 
```suggestion
* `max_tasks_per_node` - (Optional) Specifies the maximum number of tasks that can run concurrently on a single compute node in the pool. Defaults to `1`.
```

File: azurerm/resource_arm_batch_pool_test.go [hunk 7]
```diff
@@ -323,7 +324,8 @@ resource "azurerm_batch_pool" "test" {
   resource_group_name = "${azurerm_resource_group.test.name}"
   account_name        = "${azurerm_batch_account.test.name}"
   display_name        = "Test Acc Pool"
-  vm_size             = "Standard_A1"
+	vm_size             = "Standard_A1",
```
Comment (katbyte) [inline azurerm/resource_arm_batch_pool_test.go]: Could we fix the formatting/alignment here?

File: website/docs/r/batch_pool.html.markdown (hunk 6 above)
Comment (stuartleeks) [reply website/docs/r/batch_pool.html.markdown]: Yup - added to both docs

File: azurerm/resource_arm_batch_pool_test.go (hunk 7 above)
Comment (stuartleeks) [reply azurerm/resource_arm_batch_pool_test.go]: Done

**Comment by stuartleeks** [comment]
Thanks @katbyte - I've made the changes you requested.Let me know if you have any further comments

File: website/docs/d/batch_pool.html.markdown [hunk 8]
```diff
@@ -43,6 +43,8 @@ The following attributes are exported:
 
 * `start_task` - A `start_task` block that describes the start task settings for the Batch pool.
 
+* `max_tasks_per_node` - The maximum number of tasks that can run concurrently on a single compute node in the pool. Defaults to `1`.
```
Comment (katbyte) [inline website/docs/d/batch_pool.html.markdown]: The default only needs to be on the resource 😅 

```suggestion
* `max_tasks_per_node` - The maximum number of tasks that can run concurrently on a single compute node in the pool.
```

**Pull request merged**